component/prometheus/integration/integrationComponent.go

```diff
@@ -0,0 +1,119 @@
+package integration
+
+import (
+	"context"
+	"net/http"
+	"path"
+	"sync"
+
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/discovery"
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/prometheus/common/model"
+)
+
+// Creator is a function provided by an implementation to create a concrete integration instance.
+type Creator func(component.Options, component.Arguments) (integrations.Integration, error)
+
+// Exports are simply a list of targets for a scraper to consume.
+type Exports struct {
+	Targets []discovery.Target `river:"targets,attr"`
+}
+
+type Component struct {
+	opts component.Options
+
+	mut sync.Mutex
+
+	reload chan struct{}
+
+	creator Creator
+
+	integration    integrations.Integration
+	metricsHandler http.Handler
+}
+
+func New(creator Creator, name string) func(component.Options, component.Arguments) (component.Component, error) {
+	return func(opts component.Options, args component.Arguments) (component.Component, error) {
+		c := &Component{
+			opts:    opts,
+			reload:  make(chan struct{}, 1),
+			creator: creator,
+		}
+		// Call to Update() to set the output once at the start.
+		if err := c.Update(args); err != nil {
+			return nil, err
+		}
+		targets := []discovery.Target{{
+			model.AddressLabel:     opts.HTTPListenAddr,
+			model.SchemeLabel:      "http",
+			model.MetricsPathLabel: path.Join(opts.HTTPPath, "metrics"),
+			"name":                 name,
+		}}
```

Comment:
Reminder about #2424, it's not a blocker for this PR but it's still an open question for these integrations.

Reply:
Thanks for reminder. I will think it over and make sure we finalize before merging.